InvenTree/part/test_api.py

```diff
@@ -403,6 +404,54 @@ class Target(IntEnum):
                     child.refresh_from_db()
                     self.assertEqual(child.parent, parent_category)
 
+    def test_structural(self):
+        """Test the effectiveness of structural categories
+
+        Make sure:
+        - Parts cannot be created in structural categories
+        - Parts cannot be assigned to structural categories
+        """
+        structural_category = PartCategory.objects.create(
+            name='Structural category',
+            description='This is the structural category',
+            parent=None,
+            structural=True
+        )
+
+        part_count_before = Part.objects.count()
+        try:
+            part = Part.objects.create(
+                name="Part which shall not be created",
+                description="-",
+                category=structural_category
+            )
+        except ValidationError:
+            pass
+
+        self.assertEqual(part_count_before, Part.objects.count())
+
+        non_structural_category = PartCategory.objects.create(
+            name='Non-structural category',
+            description='This is a non-structural category',
+            parent=None,
+            structural=False
+        )
+
+        part = Part.objects.create(
+            name="Part which category will be changed to structural",
+            description="-",
+            category=non_structural_category
+        )
+
+        part.category = non_structural_category
+
+        try:
+            part.save()
+        except ValidationError:
+            pass
```

Comment:
Question(blocking): this Error is not raised during the tests (see test coverage); why is it here? If it should be raises I would suggest using self.assertRaises and to remove it if that is not the case